### ISSUE TITLE
Do not automatically update user's status from Offline

### ISSUE THREAD

**Issue by DHaussermann** [mattermost/mattermost-plugin-mscalendar#125]
There are a couple design gaps with the current implementation of status change. When the status is set to change without user confirmation there are issues.

1. ~~When confirm status change is set to `No` and the user is `offline` we should not change the users status. There does not need to be any change in behavior when confirm status change is set to `Yes`.~~ Fixed by  @larkox in a separate PR
2. When a users status is updated by the job there are cases where we should store the previous status and return the user to that status rather than `online`

**Comment by larkox** [comment]
@aaronrothschild @asaadmahmood 
To tackle 2 we should consider several cases. Let's start defining the configurations:
1. User can decide to update the status automatically (update status) or to get confirmation before a status update (get confirmation)
2. User can decide to move to away during meetings (to away) or to move to DnD (to dnd).

Case A:
1. User is configured as Update Status and to DnD
2. The user has been automatically set to away because he was inactive
3. When the meeting finishes:
3.1 Should the plugin set him to away? This will keep his status as away even if he is actively using the app.

Case B:
1. User is configured as Get confirmation and to DnD
2. User is away when getting the confirmation message
3. User is online when hitting the Yes button on the confirmation message.
4. When the meeting finishes:
4.1: Should he be returned to away, as it was when he received the message?
4.2: Should he be returned to online, as it was when he changed the status?

Case C:
1. User is configured as get confirmation and to DnD
2. User has manually set himself as offline
3. User hits the yes button on the confirmation message while still offline.
4. When the meeting finishes:
4.1: Should he be returned to offline?
4.2: Should he be returned to online?

Case D:
1. User is configured as get confirmation and to Away
2. User is DnD
3. When the meeting finishes:
3.1: Should he be returned to Away?

Case E:
1. User is configured as update status and to DnD
2. User is Away
3. During the meeting, the user changes from DnD to Online
4. When the meeting finishes:
4.1: Should he be returned to Away?
4.2: Should the user be left as Online?

I think that there are too many edge cases to handle this beautifully. Current implementation would behave like this:
- Any status different to the one selected by the user (dnd or away) is considered a "free status".
- The status selected by the user is considered a "busy status".
- When I finish the meeting, if I am in a free status, I do nothing.
- When I finish the meeting, if I am in the busy status, I change to online.

**Comment by asaadmahmood** [comment]
@larkox 
Before starting, we keep this principle in mind:
**If a user changes his status, then the plugin should not revert his status.**

Case A
- No, because based on our principle, he (or his activity) changed his status in the principle above.

Case B
- Returned to online. As he was online when he changed his status (he'll go away if he remains offline in a while).

Case C
- Offline

Case D
- If the user was DND before the meeting, then he should be marked DND after the meeting.

Case E
- User changed his status, so reverting should not happen, he will be left Online.

**Comment by mickmister** [comment]
If the user's was manually set before the meeting, then I agree that we should set it back to that status. We can know if this is the case if `status.Manual == true` in the server's status data model. I also agree that if the user changes their status during their meeting, that we should leave it alone. It's as if the meeting did not happen.

But if `status.Manual == false` (the user is either `Online`, `Away`, or `Offline` due to the system adjusting based on activity), then I believe we should set the user to `Online`, as it is the only way to set the server's status management back in play, and will eventually set them to `Away`/`Offline` if they have been inactive for a certain amount of time, as it should. Any change explicitly done by the user or the bot is counted as `Manual == true`, but setting to `Online` is only a temporary `Manual == true` status change.

@asaadmahmood The main thing I'm trying to cover here, is that if the user was "locked" into a status that they set manually, then we should obviously "lock" them into the same status again after the meeting. But if the user's previous status was not manually set, we should not "lock" them into their previous status. Currently the only way to not "lock" them in is to set them to `Online`. The `Online` status will be changed soon after by the system if they are inactive.
